crowbar_framework/app/models/neutron_service.rb

```diff
@@ -104,6 +106,50 @@ def create_proposal
     base
   end
 
+  def validate_gre gre_settings
+    if gre_settings["tunnel_id_start"] < 1 || gre_settings["tunnel_id_start"] > 2147483647
+      validation_error("Start of GRE tunnel ID range must be between 0 and 2147483647")
+    end
+    if gre_settings["tunnel_id_end"]  < 1 || gre_settings["tunnel_id_end"] > 2147483647
+      validation_error("End of GRE tunnel ID range must be between 0 and 2147483647")
+    end
+    if gre_settings["tunnel_id_start"] > gre_settings["tunnel_id_end"]
+      validation_error("End of GRE tunnel ID range must be higher than start of GRE tunnel ID range")
+    elsif gre_settings["tunnel_id_start"] + 1 > gre_settings["tunnel_id_end"]
```

Comment:
Hrm, no, it means that we do not accept `gre_settings["tunnel_id_start"] == gre_settings["tunnel_id_end"]`. The reason I did two tests is really for the different error messages.